src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs

```diff
@@ -395,6 +395,40 @@ private void DynamoView_Loaded(object sender, EventArgs e)
             watchSettingsControl.DataContext = background_preview;
         }
 
+        /// <summary>
+        /// Call this method to optionally bring up terms of use dialog. User 
+        /// needs to accept terms of use before any packages can be downloaded 
+        /// from package manager.
+        /// </summary>
+        /// <returns>Returns true if the terms of use for downloading a package 
+        /// is accepted by the user, or false otherwise. If this method returns 
+        /// false, then download of package should be terminated.</returns>
+        /// 
+        bool DisplayTermsOfUseForAcceptance()
+        {
+            var prefSettings = dynamoViewModel.Model.PreferenceSettings;
+            if (prefSettings.PackageDownloadTouAccepted)
+                return true; // User accepts the terms of use.
+
+
+            TermsOfUseView termsOfUseView = new TermsOfUseView(new TermsOfUseViewModel())
+                {
+                    Owner = this,
+                    WindowStartupLocation = WindowStartupLocation.CenterOwner
+                };
+            termsOfUseView.ShowDialog();
+
+            bool termsOfUseAccepted = termsOfUseView.AcceptedTermsOfUse;
+
+            if (termsOfUseAccepted)
```

Comment:
I think we don't need to check here, simply set this:

`prefSettings.PackageDownloadTouAccepted = termsOfUseAccepted `